core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassService.scala

```diff
@@ -0,0 +1,444 @@
+        } yield ()
+      } else {
+        logger.info(
+          s"Not enough IAM Policy Role Binding quota available to add FastPass access for ${ctx.userInfo.userEmail.value} in workspace ${workspace.toWorkspaceName}"
+        )
+        DBIO.successful()
+      }
+    }
+  }
+
+  def setupFastPassForUserInClonedWorkspace(parentWorkspace: Workspace,
+                                            childWorkspace: Workspace
+  ): ReadWriteAction[Unit] = {
+    if (!config.enabled) {
+      logger.debug(s"FastPass is disabled. Will not grant FastPass access to ${parentWorkspace.toWorkspaceName}")
+      return DBIO.successful()
+    }
+
+    DBIO.from(quotaAvailableForClonedWorkspaceFastPassGrants(parentWorkspace, childWorkspace)).flatMap {
+      quotaAvailable =>
+        if (quotaAvailable) {
+          logger.info(
+            s"Adding FastPass access for ${ctx.userInfo.userEmail} in workspace being cloned ${parentWorkspace.toWorkspaceName}"
+          )
+          val expirationDate = DateTime.now(DateTimeZone.UTC).plus(config.grantPeriod.toMillis)
+          for {
+            petEmail <- DBIO.from(samDAO.getUserPetServiceAccount(ctx, childWorkspace.googleProjectId))
+            userAndPet = UserAndPetEmails(WorkbenchEmail(ctx.userInfo.userEmail.value), petEmail)
+            _ <- setupBucketRoles(parentWorkspace, Set(SamWorkspaceRoles.reader), userAndPet, expirationDate)
+            _ <- DBIO
+              .from(openTelemetry.incrementCounter("fastpass-granted-user", tags = openTelemetryTags).unsafeToFuture())
+          } yield ()
+        } else {
+          logger.info(
+            s"Not enough IAM Policy Role Binding quota available to add FastPass access for ${ctx.userInfo.userEmail.value} in parent workspace ${parentWorkspace.toWorkspaceName}"
+          )
+          DBIO.successful()
+        }
+    }
+  }
+
+  def removePolicyFastPassesForUser(workspace: Workspace,
+                                    policyName: SamResourcePolicyName,
+                                    email: String
+  ): ReadWriteAction[Unit] =
+    for {
+      maybeUserSubjectId <- DBIO.from(samDAO.getUserIdInfo(email, ctx).map {
+        case SamDAO.User(x) => x.googleSubjectId
+        case _              => None
+      })
+      if maybeUserSubjectId.isDefined
+      userSubjectId = maybeUserSubjectId.orNull
+      samPolicy <- DBIO.from(samDAO.getPolicy(SamResourceTypeNames.workspace, workspace.workspaceId, policyName, ctx))
+      organizationRoles = samPolicy.roles.flatMap(samWorkspaceRoleToGoogleProjectIamRoles) ++ samPolicy.roles.flatMap(
+        samWorkspaceRolesToGoogleBucketIamRoles
+      )
+      fastPassGrants <- dataAccess.fastPassGrantQuery.findFastPassGrantsForUserInWorkspace(
+        workspace.workspaceIdAsUUID,
+        RawlsUserSubjectId(userSubjectId)
+      )
+      userAndRoleSpecificFastPassGrants = fastPassGrants.filter(grant =>
+        organizationRoles.contains(grant.organizationRole)
```

Comment:
What rules are we filtering out here?

Reply:
Agreed - suggest adding a comment that describes what this thing needs to be doing in simple english

Reply:
Added a clarifying comment